pkg/chain/ethereum/tbtc.go

```diff
@@ -996,6 +997,125 @@ func (tc *TbtcChain) OnHeartbeatRequested(
 	})
 }
 
+func (tc *TbtcChain) PastDepositRevealedEvents(
+	filter *tbtc.DepositRevealedEventFilter,
+) ([]*tbtc.DepositRevealedEvent, error) {
+	var startBlock uint64
+	var endBlock *uint64
+	var depositor []common.Address
+	var walletPublicKeyHash [][20]byte
+
+	if filter != nil {
+		startBlock = filter.StartBlock
+		endBlock = filter.EndBlock
+
+		for _, d := range filter.Depositor {
+			depositor = append(depositor, common.HexToAddress(d.String()))
+		}
+
+		walletPublicKeyHash = filter.WalletPublicKeyHash
+	}
+
+	events, err := tc.bridge.PastDepositRevealedEvents(
+		startBlock,
+		endBlock,
+		depositor,
+		walletPublicKeyHash,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	convertedEvents := make([]*tbtc.DepositRevealedEvent, 0)
+	for _, event := range events {
+		var vault *chain.Address
+		if event.Vault != [20]byte{} {
+			v := chain.Address(event.Vault.Hex())
+			vault = &v
+		}
```

Comment:
Wouldn't it be easier to call `event.Vault.Hex()` and let it be zero-bytes? 

The client will have to compare the address against `TBTCVault` contract address anyway.

Same question to the conversion in `GetDepositRequest`.

Reply:
> Wouldn't it be easier to call event.Vault.Hex() and let it be zero-bytes?

Actually, this is a matter of how we define an optional value. I think it's more intuitive to have a pointer that explicitly indicates the value can be `nil` than a `chain.Address` which can be empty or zeroed. Remember that `chain.Address` is a chain-agnostic type so 20 zeroed bytes can be a valid value for Ethereum but this may be not the case for other chains. If we use a pointer, the client does not need to know what value represents an empty address for the given chain. It just checks whether the value is `nil` or not.

> The client will have to compare the address against TBTCVault contract address anyway.

I don't see any place where the client will need to compare this with the `TBTCVault` at the moment. Do you have something particular in mind?

Reply:
> Actually, this is a matter of how we define an optional value. I think it's more intuitive to have a pointer that explicitly indicates the value can be nil than a chain.Address which can be empty or zeroed. Remember that chain.Address is a chain-agnostic type so 20 zeroed bytes can be a valid value for Ethereum but this may be not the case for other chains. If we use a pointer, the client does not need to know what value represents an empty address for the given chain. It just checks whether the value is nil or not.

Good reasoning. Can we add a code comment to `DepositRevealedEvent` and `DepositChainRequest` in `pkg/tbtc/chain.go` and make it an API requirement that the `Vault` should be `nil` if it wasn't set on the chain?

> I don't see any place where the client will need to compare this with the TBTCVault at the moment. Do you have something particular in mind?

You are right. The `WalletCoordinator` contract checks if all deposits [target the same vault](https://github.com/keep-network/tbtc-v2/blob/031249d0001ab065e80a48e331da1155a636b9e8/solidity/contracts/bridge/WalletCoordinator.sol#L532-L540) and this is enough.

Reply:
See 0bfa45aa